docs/subnets/introduction-to-vm.md

```diff
@@ -0,0 +1,361 @@
+	//                 engine that manages this VM. The write lock is held
+	//                 whenever code in the consensus engine calls the VM.
+	// [dbManager]: The manager of the database this VM will persist data to.
+	// [genesisBytes]: The byte-encoding of the genesis information of this
+	//                 VM. The VM uses it to initialize its state. For
+	//                 example, if this VM were an account-based payments
+	//                 system, `genesisBytes` would probably contain a genesis
+	//                 transaction that gives coins to some accounts, and this
+	//                 transaction would be in the genesis block.
+	// [toEngine]: The channel used to send messages to the consensus engine.
+	// [fxs]: Feature extensions that attach to this VM.
+	Initialize(
+		ctx *snow.Context,
+		dbManager manager.Manager,
+		genesisBytes []byte,
+		upgradeBytes []byte,
+		configBytes []byte,
+		toEngine chan<- Message,
+		fxs []*Fx,
+		appSender AppSender,
+	) error
+
+	// Bootstrapping is called when the node is starting to bootstrap this chain.
+	Bootstrapping() error
+
+	// Bootstrapped is called when the node is done bootstrapping this chain.
+	Bootstrapped() error
+
+	// Shutdown is called when the node is shutting down.
+	Shutdown() error
+
+	// Version returns the version of the VM this node is running.
+	Version() (string, error)
+
+	// Creates the HTTP handlers for custom VM network calls.
+	//
+	// This exposes handlers that the outside world can use to communicate with
+	// a static reference to the VM. Each handler has the path:
+	// [Address of node]/ext/VM/[VM ID]/[extension]
+	//
+	// Returns a mapping from [extension]s to HTTP handlers.
+	//
+	// Each extension can specify how locking is managed for convenience.
+	//
+	// For example, it might make sense to have an extension for creating
+	// genesis bytes this VM can interpret.
+	CreateStaticHandlers() (map[string]*HTTPHandler, error)
+
+	// Creates the HTTP handlers for custom chain network calls.
+	//
+	// This exposes handlers that the outside world can use to communicate with
+	// the chain. Each handler has the path:
+	// [Address of node]/ext/bc/[chain ID]/[extension]
+	//
+	// Returns a mapping from [extension]s to HTTP handlers.
+	//
+	// Each extension can specify how locking is managed for convenience.
+	//
+	// For example, if this VM implements an account-based payments system,
+	// it have an extension called `accounts`, where clients could get
+	// information about their accounts.
+	CreateHandlers() (map[string]*HTTPHandler, error)
+}
+```
+
+### `snowman.Block`
+
+You may have noticed the `snowman.Block` type referenced in the `block.ChainVM` interface. It describes the methods that a block must implement to be a block in a linear (Snowman) chain.
+
+Let’s look at this interface and its methods. You can see the full file from [here.](https://github.com/ava-labs/avalanchego/blob/v1.7.4/snow/consensus/snowman/block.go)
```

Comment:
Should the link to the master branch or v1.7.4 branch? If master branch, please search and replace in the doc.